packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1817,6 +1833,9 @@ export function startGestureTransition(
       }
       for (let i = 0; i < animations.length; i++) {
         const anim = animations[i];
+        if (anim.playState !== 'running') {
```

Comment:
This alone doesn't really help anything because the "finished" animations seems to interfere regardless but we shouldn't really restart any animation here. We expect all of them to be running. So this is just defensive coding.